### PR TITLE
Updated the door-lock-cluster.xml door-lock-cluster.xml to use the sp…

### LINKED ISSUE #15528: App cluster XMLs should use original type definitions instead of BITMAP* directly
#### Problem

Various app cluster XMLs define bitmaps with their underlying enum-like definitions:

e.g
```
    <bitmap name="WiFiSecurity" type="BITMAP8">
        <field name="Unencrypted" mask="0x1"/>
        <field name="WEP-PERSONAL" mask="0x2"/>
        <field name="WPA-PERSONAL" mask="0x4"/>
        <field name="WPA2-PERSONAL" mask="0x8"/>
        <field name="WPA3-PERSONAL" mask="0x10"/>
    </bitmap>
```

However, they actually lose the type information when instantiating these in various instances:
```
    <struct name="WiFiInterfaceScanResult">
        <cluster code="0x0031"/>
        <item name="Security" type="BITMAP8"/> <---- should be of type `WifiSecurity` instead.
        <item name="SSID" type="OCTET_STRING"/>
        <item name="BSSID" type="OCTET_STRING"/>
        <item name="Channel" type="INT16U"/>
        <item name="WiFiBand" type="WiFiBand"/>
        <item name="RSSI" type="INT8S"/>
    </struct>
```

##### Solution

Update the various app cluster XMLs to use the original type that is eventually backed by a BITMAP*

### PR DESCRIPTION
**Pull request by hubTab**
Updated the DoorLock Cluster XML to use type definitions.

#### Problem
What is being fixed?
* The door-lock-cluster.xml should use the specifications type definitions, instead of using the enum* directly. Partially Fixes #15528, by updating the door lock cluster xml file to use the specifications type definitions.

#### Change overview
What's in this PR
* Updated the door-lock-cluster.xml door-lock-cluster.xml to use the specifications type definitions, instead of using the enum* directly.
* Also, fixed the comment in the ManualOperationEventMask attribute bit meaning set.

#### Testing
How was this tested? (at least one bullet point required)
* Regenerated all ZAP cluster code, by running   ./scripts/tools/zap_regen_all.py. Compared the generated code versus the original zzz_generated. The differences are highlighted in the attached zip file, all indicating the expected type changes. 
[CSA-15528-Test-Report.zip](https://github.com/project-chip/connectedhomeip/files/8143474/CSA-15528-Test-Report.zip)

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/door-lock-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/door-lock-cluster.xml:112]: This does not match the spec.  In the spec this is a map8, not an enum, and the values of the enum are wrong here.

File: src/app/zap-templates/zcl/data-model/chip/door-lock-cluster.xml
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/door-lock-cluster.xml:145]: In the spec these are just map16.  Do we need a spec issue to add the named bitmap types?  Similar for other bits below.

File: src/app/zap-templates/zcl/data-model/chip/door-lock-cluster.xml
Comment (hubTab) [reply src/app/zap-templates/zcl/data-model/chip/door-lock-cluster.xml:112]: Good catch. Will add DlCredentialRuleMask bitmap and update CredentialRulesSupport's type to DlCredentialRuleMask.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
Fast tracking, given this has had enough time for review.

@hubTab Can you please regen zap?

**Comment by hubTab** [comment]
@woody-apple ,
This PR is dependent on PR #15697. Waiting for it be merged into master, to be able to regen zap. (scripts/examples/gn_build_example.sh examples/all-clusters-app/linux out/debug already verified)

**Comment by hubTab** [comment]
Just confirming that as PR #15697 has been merged into master , I am working on rebasing this PR on master, and verify it checks running the two tests, before merging this PR into master.

**Comment by hubTab** [comment]
@woody-apple , @bzbarsky-apple , @mrjerryjohns ,

This PR has been rebased on the master branch, which now includes the #15697 fix. I confirm that this PR successfully built both the all-cluster-app and the chip-tool apps. In addition it includes the regenerated the zzz code, being ready to be merged in back into master. Since I do not have the rights to proceed with the merge, can one of you please help merge it in?

Thanks!
Eugen

**Review by mrjerryjohns — APPROVED** [review]

**Pull request merged**
